### ISSUE TITLE
Rule "curly" with "multi" has false positive on nested if-for-if, fixer changes code behavior

### ISSUE THREAD

**Issue by drakedevel** [eslint/eslint#12928]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** 12.16.0
* **npm Version:** 6.13.4

This bug also reproduces on ESLint 7.0.0-alpha.1.

**What parser (default, Babel-ESLint, etc.) are you using?**
Default parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "rules": {
    "curly": ["warn", "multi"]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var x = [''];
if (x)
  for (var i in x) {
    if (i > 0)
      console.log(i);
  }
else
  console.log('whoops');
```

```bash
./node_modules/.bin/eslint --fix index.js
```

**What did you expect to happen?**
ESLint should make no changes to the code, as the only curly braces present are required.

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint changes the code to the following:

```js
var x = [''];
if (x)
  for (var i in x) 
    if (i > 0)
      console.log(i);
  
else
  console.log('whoops');
```

This is incorrect -- the `else` is now associated with the inner `if` instead of the outer one. The execution behavior changes accordingly:

```shell
$ node index.js 
$ ./node_modules/.bin/eslint index.js

/tmp/eslint-repro/index.js
  3:3  warning  Unnecessary { after 'for-in'  curly

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

$ ./node_modules/.bin/eslint --fix index.js 
$ node index.js 
whoops
```

I did a little bit more experimentation: when I replace the `for` block with a `while` block, the same issue occurs. However, when I replace the `for` block with another `if`, ESLint correctly detects that the curly braces are mandatory.

**Are you willing to submit a pull request to fix this bug?**
I can if necessary, although the fix is probably much faster for somebody who's already familiar with the relevant code.

**Comment by mdjermanovic** [comment]
Hi @drakedevel, thanks for the bug report!

Confirmed:

* [Eslint Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwiY3VybHlcIjogW1wid2FyblwiLCBcIm11bHRpXCJdICovXG5cbnZhciB4ID0gWycnXTtcbmlmICh4KVxuICBmb3IgKHZhciBpIGluIHgpIHtcbiAgICBpZiAoaSA+IDApXG4gICAgICBjb25zb2xlLmxvZyhpKTtcbiAgfVxuZWxzZVxuICBjb25zb2xlLmxvZygnd2hvb3BzJyk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) 
* [ASTExplorer with the original code](https://astexplorer.net/#/gist/165bcee5a5d3405a66c02cce7de0e7f4/ec557f3291b26260ece8d43757e0e5d07e570fcb)
* [ASTExplorer with the fixed code](https://astexplorer.net/#/gist/165bcee5a5d3405a66c02cce7de0e7f4/a46bff6b89eb9af6e4fda9359ced4d13d112b5b2)

The outer `if` has `alternate: null` in AST for the fixed code.

**Comment by mdjermanovic** [comment]
I'm working on this.

**Comment by drakedevel** [comment]
Confirmed fixed for me in 7.0.0-alpha.2, thanks!
